pkg/operation/hybridbotanist/cloud_config.go

```diff
@@ -165,19 +167,32 @@ func (b *HybridBotanist) generateOriginalConfig() (map[string]interface{}, error
 	return b.InjectShootShootImages(originalConfig, common.HyperkubeImageName, common.PauseContainerImageName)
 }
 
-func (b *HybridBotanist) computeOperatingSystemConfigsForWorker(machineTypes []gardenv1beta1.MachineType, machineImageName string, downloaderConfig, originalConfig map[string]interface{}, worker gardenv1beta1.Worker) (*shoot.CloudConfig, error) {
+func (b *HybridBotanist) computeOperatingSystemConfigsForWorker(machineTypes []gardenv1beta1.MachineType, machineImage *gardenv1beta1.MachineImage, downloaderConfig, originalConfig map[string]interface{}, worker gardenv1beta1.Worker) (*shoot.CloudConfig, error) {
 	var (
 		evictionHardMemoryAvailable, evictionSoftMemoryAvailable = getEvictionMemoryAvailable(machineTypes, worker.MachineType)
 		secretName                                               = b.Shoot.ComputeCloudConfigSecretName(worker.Name)
 	)
 
 	downloaderConfig["secretName"] = secretName
+
+	var  customization  = map[string]interface{}{}
+	if machineImage.ProviderConfig != nil {
+		err := yaml.Unmarshal(machineImage.ProviderConfig.Raw, &customization)
```

Comment:
What's the reason to unmarshal this? can't you simply "copy" it blindly to the `providerConfig` field of the `OperatingSystemConfig` resource?

Reply:
I think that in order to be used in the helm template it must be unmarshaled.

Reply:
Hm yeah, true. Maybe you can change the code such that it doesn't render the Helm chart for creating the `OperatingSystemConfig` resource but create it directly via Go means?

For example, when Gardener creates the `Infrastructure` (or, basically, every other extension CRD) it does that directly via the Go type, see https://github.com/gardener/gardener/blob/master/pkg/operation/botanist/infrastructure.go#L50-L90.

Then we don't need to unmarshal this, I guess that's cleaner and more consistent then.

Reply:
I just saw helm has the `toYaml` function maybe this could be used with the unmarshalled ProviderConfig:
```
{{  toYaml .Values.osc.providerConfig | indent 2 }}
```

Reply:
Maybe, but I guess it's still cleaner to do it within Go (mainly, because all other extensions are also created/updated this way).

Reply:
> For example, when Gardener creates the Infrastructure (or, basically, every other extension CRD) it does that directly via the Go type

I think this case is different because of the complexity of the OSC. I'm not sure that saving this unmarshalling worth the additional complexity. In any case, I would explore the idea of using `toYaml` function I mentioned above. Need to document about it.

Reply:
OK, agreed, let's do it in a separate PR.